modules/custom/cu_core/cu_core.install

```diff
@@ -5,44 +5,24 @@
  * Install hooks for cu_core module.
  */
 
-/**
- * Implemenets hook_enable().
- */
-function cu_core_enable() {
-
-}
 
 /**
  * Implements hook_install().
  */
 function cu_core_install() {
 
-  // Check for environmental variables to install other cores.
-  $check = cu_core_environment_check();
+  $hosting = cu_core_get_hosting();
 
-  // If needed, exit install before something else happens.
-  if ($check['exit?']) {
-    return;
+  if ($hosting == 'travis') {
+    module_enable(array('travis_hosting'));
   }
 
-  // If no other special environment found, install default modules.
-  if ($check['no_environment?']) {
-    $modules = array(
-      'cu_atlas',
-      'atlas',
-      'atlas_statistics',
-      'noreqnewpass',
-      'ldap_authentication',
-      'ldap_servers',
-      'ldap_user',
-      'cu_ldap',
-      'cu_users',
-      'express_user_lock',
-      'memcache',
-      'varnish'
-      // Adding caching modules here for changes involving Travis CI testing.
-    );
-    module_enable($modules);
+  if ($hosting == 'pantheon') {
+    module_enable(array('pantheon_hosting'));
+  }
+
+  if ($hosting == FALSE) {
+    module_enable(array('ng_hosting'));
```

Comment:
Later in the code modules get enabled that should go into ng_hosting.  

```php

  // these modules need to be enabled late in the install process and/or to be ignored by profile_module_manager's report.
  $module_array = array(
    'cu_users',
    'memcache',
    'varnish'  // Adding caching modules here for changes involving Travis CI testing.
  );
```

Reply:
varnish and memcache can be hard dependencies of the hosting module.  cu_users needs to be enabled late in the install process because it is what forces secure_permissions to be enabled and that dramatically slows down the install process.  Not sure if enabling that an install hook of the core is late enough.